Build failure: reiser4progs and partitionmanager after staging-next 2024-12-04 #361878

## Steps To Reproduce

Steps to reproduce the behavior:

1. build reiser4progs or partitionmanager at nixpkgs/master

## Build log

```
error: builder for '/nix/store/8y7805npjq8h21khs1qhjkm3wd2y5hzp-reiser4progs-2.0.5.drv' failed with exit code 2;
       last 25 log lines:
       >   146 |                           "(%llu). Use -f to force over.",
       >   147 |                           result, block_count);
       >       |                                   ~~~~~~~~~~~
       >       |                                   |
       >       |                                   uint64_t {aka long unsigned int}
       > volume.c: At top level:
       > volume.c:174:31: error: initialization of 'int (*)(uint64_t *, uint64_t,  uint32_t,  int,  int)' {aka 'int (*)(long unsigned int *, long unsigned int,  unsigned int,  int,  int)'} from incompatible pointer type 'int (*)(uint64_t *, uint32_t,  uint64_t,  int,  int)' {aka 'int (*)(long unsigned int *, unsigned int,  long unsigned int,  int,  int)'} []
       >   174 |         .advise_stripe_size = advise_stripe_size_simple,
       >       |                               ^~~~~~~~~~~~~~~~~~~~~~~~~
       > volume.c:174:31: note: (near initialization for 'simple_vol_plug.advise_stripe_size')
       > volume.c:177:34: error: initialization of 'uint64_t (*)(uint64_t,  int)' {aka 'long unsigned int (*)(long unsigned int,  int)'} from incompatible pointer type 'uint64_t (*)(uint64_t)' {aka 'long unsigned int (*)(long unsigned int)'} []
       >   177 |         .default_data_capacity = default_data_capacity_simple
       >       |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > volume.c:177:34: note: (near initialization for 'simple_vol_plug.default_data_capacity')
       > volume.c:186:31: error: initialization of 'int (*)(uint64_t *, uint64_t,  uint32_t,  int,  int)' {aka 'int (*)(long unsigned int *, long unsigned int,  unsigned int,  int,  int)'} from incompatible pointer type 'int (*)(uint64_t *, uint32_t,  uint64_t,  int,  int)' {aka 'int (*)(long unsigned int *, unsigned int,  long unsigned int,  int,  int)'} []
       >   186 |         .advise_stripe_size = advise_stripe_size_asym,
       >       |                               ^~~~~~~~~~~~~~~~~~~~~~~
       > volume.c:186:31: note: (near initialization for 'asym_vol_plug.advise_stripe_size')
       > make[3]: *** [Makefile:419: libvolume_static_la-volume.lo] Error 1
       > make[3]: Leaving directory '/build/reiser4progs-2.0.5/plugin/vol'
       > make[2]: *** [Makefile:613: all-recursive] Error 1
       > make[2]: Leaving directory '/build/reiser4progs-2.0.5/plugin'
       > make[1]: *** [Makefile:463: all-recursive] Error 1
       > make[1]: Leaving directory '/build/reiser4progs-2.0.5'
       > make: *** [Makefile:372: all] Error 2
       For full logs, run 'nix log /nix/store/8y7805npjq8h21khs1qhjkm3wd2y5hzp-reiser4progs-2.0.5.drv'.
error: 1 dependencies of derivation '/nix/store/5pyrc6y1ri4f92sf067qbkckdyipa64n-partitionmanager-23.08.5.drv' failed to build
```

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

@peterhoeg
@oxalica

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/doc/manual/release-notes/rl-2505.section.md
+++ b/nixos/doc/manual/release-notes/rl-2505.section.md
@@ -290,6 +290,11 @@
 
   To get the necessary hash of the vendored dependencies, omit `hash`. The build will fail and tell you the correct value.
 
+- KDE Partition Manager `partitionmanager`'s support for ReiserFS is removed.
+  ReiserFS has not been actively maintained for many years. It has been marked as obsolete since Linux 6.6, and
+  [is removed](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=c01f664e4ca210823b7594b50669bbd9b0a3c3b0)
+  in Linux 6.13.
+
 - `programs.fzf.keybindings` now supports the fish shell.
 
 <!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->

--- a/pkgs/applications/kde/partitionmanager/default.nix
+++ b/pkgs/applications/kde/partitionmanager/default.nix
@@ -24,8 +24,6 @@
   jfsutils,
   nilfs-utils,
   ntfs3g,
-  reiser4progs,
-  reiserfsprogs,
   udftools,
   xfsprogs,
   zfs,
@@ -54,8 +52,7 @@ let
     jfsutils
     nilfs-utils
     ntfs3g
-    reiser4progs
-    reiserfsprogs
+    # reiser{4,fs}progs intentionally omitted due to filesystem removal from Linux.
     udftools
     xfsprogs
     zfs
@@ -98,7 +95,7 @@ mkDerivation {
     longDescription = ''
       KDE Partition Manager is a utility to help you manage the disks, partitions, and file systems on your computer.
       It allows you to easily create, copy, move, delete, back up, restore, and resize them without losing data.
-      It supports a large number of file systems, including ext2/3/4, btrfs, reiserfs, NTFS, FAT16/32, JFS, XFS and more.
+      It supports a large number of file systems, including ext2/3/4, btrfs, NTFS, FAT16/32, JFS, XFS and more.
 
       To install on NixOS, use the option `programs.partition-manager.enable = true`.
     '';

--- a/pkgs/by-name/re/reiser4progs/package.nix
+++ b/pkgs/by-name/re/reiser4progs/package.nix
@@ -38,5 +38,11 @@ stdenv.mkDerivation rec {
     description = "Reiser4 utilities";
     license = licenses.gpl2Plus;
     platforms = platforms.linux;
+
+    # error: initialization of
+    # 'int (*)(uint64_t *, uint64_t,  uint32_t,  int,  int)' {aka 'int (*)(long unsigned int *, long unsigned int,  unsigned int,  int,  int)'}
+    # from incompatible pointer type
+    # 'int (*)(uint64_t *, uint32_t,  uint64_t,  int,  int)' {aka 'int (*)(long unsigned int *, unsigned int,  long unsigned int,  int,  int)'}
+    broken = true;
   };
 }

--- a/pkgs/kde/gear/kpmcore/default.nix
+++ b/pkgs/kde/gear/kpmcore/default.nix
@@ -18,8 +18,6 @@
   jfsutils,
   nilfs-utils,
   ntfs3g,
-  reiser4progs,
-  reiserfsprogs,
   udftools,
   xfsprogs,
   zfs,
@@ -44,8 +42,7 @@ let
     jfsutils
     nilfs-utils
     ntfs3g
-    reiser4progs
-    reiserfsprogs
+    # reiser{4,fs}progs intentionally omitted due to filesystem removal from Linux.
     udftools
     xfsprogs
     zfs

